dsgrid/registry/registry_manager.py

```diff
@@ -524,24 +525,26 @@ def copy(src: Path, dst: Path, make_data_symlinks=False, use_rsync=False, force=
         if not {x.name for x in src.iterdir()}.issuperset({"configs", "data"}):
             raise DSGInvalidParameter(f"{src} is not a valid registry")
 
-        if use_rsync:
+        if mode == "rsync":
             dst.mkdir(exist_ok=True)
             cmd = f"rsync -a {src}/ {dst}"
             logger.info("rsync data with [%s]", cmd)
             check_run_command(cmd)
-        else:
+        elif mode in ("copy", "data-symlinks"):
             if dst.exists():
                 if force:
                     shutil.rmtree(dst)
                 else:
                     raise DSGInvalidParameter(f"{dst} already exists.")
             logger.info("Copy data from source registry %s", src)
-            if make_data_symlinks:
+            if mode == "data-symlinks":
                 (dst).mkdir()
                 shutil.copytree(src / "configs", dst / "configs")
                 _make_data_symlinks(src, dst)
             else:
                 shutil.copytree(src, dst, symlinks=True)
+        else:
+            raise DSGInvalidParameter(f"mode={mode} is not supported")
```

Comment:
optional - could be cleaner if you move this mode validation to `make_filtered_registry`

Reply:
click does the validation in the CLI handler. This is independent and needed extra validation.